// the-platform/useDeviceOrientation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useDeviceOrientation = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var useDeviceOrientation = function useDeviceOrientation() {
  var _React$useState = _react.default.useState({
    alpha: null,
    beta: null,
    gamma: null,
    absolute: false
  }),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      orientation = _React$useState2[0],
      setOrientation = _React$useState2[1];

  var handle = function handle(e) {
    setOrientation({
      beta: e.beta,
      alpha: e.alpha,
      gamma: e.gamma,
      absolute: e.absolute
    });
  };

  _react.default.useEffect(function () {
    window.addEventListener('deviceorientation', handle);
    return function () {
      window.removeEventListener('deviceorientation', handle);
    };
  }, []);

  return orientation;
};

exports.useDeviceOrientation = useDeviceOrientation;
